### PR TITLE
Feature: Allow deleting of local Git branches

### LINKED ISSUE #13592: Feature: Allow deleting of local Git branches
### What feature or improvement do you think would benefit Files?

*

### Requirements

- Add a delete button next to the local branch
- Show a delete branch confirmation content dialog

### Files Version

2.5.30.0

### Windows Version

10.22621.2428

### Comments

_No response_

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #13592 

**Important**
- Backup your repo before testing this feature
- Check this PR doesn't affect performance in Git folders
  - I had to revert a change made in #13608 because `Ahead / Behind` wouldn't be update after `Pull`s (see `BaseShellPage.cs`)

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [x] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Open a git repo
   2. Click on the branch name in the status bar
   3. Click on `Delete` to delete a local branch
   4. Click on `Cancel` and make sure the branch is still there
   5. Repeat but click on `Continue` and check that the branch is deleted

**Screenshots (optional)**
image

### CODE REVIEW

**Comment by yaira2** [comment]
We still haven't finalized the design for this feature.

File: src/Files.App/UserControls/StatusBarControl.xaml [hunk 1]
```diff
@@ -350,9 +351,32 @@
...
+										</Grid.ColumnDefinitions>
+
+										<TextBlock
+											VerticalAlignment="Center"
+											Text="{x:Bind Name}"
+											TextTrimming="CharacterEllipsis" />
+										<Button
+											Grid.Column="1"
+											Click="DeleteBranch_Click"
+											Content="{helpers:ResourceString Name=Delete}"
```
Comment (yaira2) [inline src/Files.App/UserControls/StatusBarControl.xaml:373]: What do you think about using an icon instead?

Comment (ferrariofilippo) [reply src/Files.App/UserControls/StatusBarControl.xaml:373]: Good idea, I think it would look better! Something like a red bin?

Comment (yaira2) [reply src/Files.App/UserControls/StatusBarControl.xaml:373]: I'll let @mdtauk answer about the color, but as far as the icon goes, a bin is probably a good choice.

**Pull request merged**